File: bae-macos/bae/bae/Views/Components/ReleaseFactsScale.swift
import SwiftUI

/// The two sizes the release's records row is drawn at: full width in the
/// import pane, and packed into the library expansion's facts line, which
/// sits in a 320-point card rather than across a pane.
enum ReleaseFactsScale {
    case pane
    case card

    /// The gap between two record links on one row.
    var recordSpacing: CGFloat {
        switch self {
        case .pane: 14
        case .card: 12
        }
    }

    /// The gap between two rows of record links.
    var recordRowSpacing: CGFloat {
        switch self {
        case .pane: 6
        case .card: 4
        }
    }

    var recordFontSize: CGFloat {
        switch self {
        case .pane: 11.5
        case .card: 11
        }
    }
}

File: bae-avalonia/Views/Components/ReleaseFactsScale.cs
namespace Bae.Desktop;

/// <summary>
/// The two sizes a release's records row is drawn at: full width in the import
/// pane, and packed into the card the library expansion's facts line opens,
/// which sits in a 320-point card rather than across a pane.
/// </summary>
internal enum ReleaseFactsScale
{
    Pane,
    Card,
}

internal static class ReleaseFactsScaleMetrics
{
    /// <summary>The gap between two record links on one row.</summary>
    internal static double RecordSpacing(this ReleaseFactsScale scale) => scale switch
    {
        ReleaseFactsScale.Pane => 14,
        ReleaseFactsScale.Card => 12,
        _ => throw new System.ArgumentOutOfRangeException(nameof(scale), scale, null),
    };

    /// <summary>The gap between two rows of record links.</summary>
    internal static double RecordRowSpacing(this ReleaseFactsScale scale) => scale switch
    {
        ReleaseFactsScale.Pane => 6,
        ReleaseFactsScale.Card => 4,
        _ => throw new System.ArgumentOutOfRangeException(nameof(scale), scale, null),
    };

    internal static double RecordFontSize(this ReleaseFactsScale scale) => scale switch
    {
        ReleaseFactsScale.Pane => 11.5,
        ReleaseFactsScale.Card => 11,
        _ => throw new System.ArgumentOutOfRangeException(nameof(scale), scale, null),
    };
}
